src/odemis/driver/smaract.py

```diff
@@ -1256,20 +1267,16 @@ def terminate(self):
     def updateMetadata(self, md):
         super(MC_5DOF, self).updateMetadata(md)
         try:
-            value = md[model.MD_PIVOT_POS]
+            pivot = md[model.MD_PIVOT_POS]
         except KeyError:
             # there is no pivot position set
             return
 
-        if not isinstance(value, dict):
-            raise ValueError("Invalid metadata, should be a coordinate dictionary but got %s." % (value,))
+        if not isinstance(pivot, dict) and pivot.keys() == {"x", "y", "z"}:
```

Comment:
This comparison only works on Python 3. On Python 2, it will always be false because `.keys()` returns a list (instead of a kind-of set).
I'm confused why your test case didn't pick this up. Did you run all the test cases recently?